metrics/sinks/stackdriver/stackdriver.go

```diff
@@ -431,54 +395,62 @@ func CreateStackdriverSink(uri *url.URL) (core.DataSink, error) {
 	return sink, nil
 }
 
-func (sink *StackdriverSink) preprocessMemoryMetrics(metricSet *core.MetricSet) *core.MetricSet {
-	usage := metricSet.MetricValues[core.MetricMemoryUsage.MetricDescriptor.Name].IntValue
-	workingSet := metricSet.MetricValues[core.MetricMemoryWorkingSet.MetricDescriptor.Name].IntValue
-	bytesUsed := core.MetricValue{
-		IntValue: usage - workingSet,
-	}
-
-	newMetricSet := &core.MetricSet{
-		MetricValues: map[string]core.MetricValue{},
+func parseBoolFlag(opts map[string][]string, name string, targetValue *bool) error {
+	if len(opts[name]) >= 1 {
+		var err error
+		*targetValue, err = strconv.ParseBool(opts[name][0])
+		if err != nil {
+			return fmt.Errorf("%s = %s is not correct boolean value", name, opts[name][0])
+		}
 	}
+	return nil
+}
 
-	newMetricSet.MetricValues["memory/bytes_used"] = bytesUsed
+func (sink *StackdriverSink) computeDerivedMetrics(metricSet *core.MetricSet) *core.MetricSet {
+	newMetricSet := &core.MetricSet{MetricValues: map[string]core.MetricValue{}}
+	usage, usageOK := metricSet.MetricValues[core.MetricMemoryUsage.MetricDescriptor.Name]
+	workingSet, workingSetOK := metricSet.MetricValues[core.MetricMemoryWorkingSet.MetricDescriptor.Name]
 
-	memoryFaults := metricSet.MetricValues[core.MetricMemoryPageFaults.MetricDescriptor.Name].IntValue
-	majorMemoryFaults := metricSet.MetricValues[core.MetricMemoryMajorPageFaults.MetricDescriptor.Name].IntValue
+	if usageOK && workingSetOK {
+		newMetricSet.MetricValues["memory/bytes_used"] = core.MetricValue{
+			IntValue: usage.IntValue - workingSet.IntValue,
+		}
+	}
 
-	minorMemoryFaults := core.MetricValue{
-		IntValue: memoryFaults - majorMemoryFaults,
+	memoryFaults, memoryFaultsOK := metricSet.MetricValues[core.MetricMemoryPageFaults.MetricDescriptor.Name]
+	majorMemoryFaults, majorMemoryFaultsOK := metricSet.MetricValues[core.MetricMemoryMajorPageFaults.MetricDescriptor.Name]
+	if memoryFaultsOK && majorMemoryFaultsOK {
+		newMetricSet.MetricValues["memory/minor_page_faults"] = core.MetricValue{
+			IntValue: memoryFaults.IntValue - majorMemoryFaults.IntValue,
+		}
 	}
-	newMetricSet.MetricValues["memory/minor_page_faults"] = minorMemoryFaults
 
 	return newMetricSet
 }
 
-func (sink *StackdriverSink) TranslateLabeledMetric(timestamp time.Time, labels map[string]string, metric core.LabeledMetric, createTime time.Time) *sd_api.TimeSeries {
-	resourceLabels := sink.getResourceLabels(labels)
+func (sink *StackdriverSink) LegacyTranslateLabeledMetric(timestamp time.Time, labels map[string]string, metric core.LabeledMetric, createTime time.Time) *sd_api.TimeSeries {
```

Comment:
Synced offline, turns out there's no need in a method for the new resource model: old disk metrics _will not be exported_. However, it requires a mention in the PR description

Reply:
Specified this more precisely in a PR description